frame/vamm/src/tests/get_price.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{
+	mock::{ExtBuilder, TestPallet},
+	pallet::VammState,
+	tests::{min_max_reserve, Decimal, RUN_CASES},
+};
+use composable_traits::vamm::{AssetType, Vamm as VammTrait};
+use frame_support::{assert_noop, assert_ok};
+use proptest::prelude::*;
+use sp_runtime::ArithmeticError;
+
+proptest! {
+	#![proptest_config(ProptestConfig::with_cases(RUN_CASES))]
+	#[test]
+	#[allow(clippy::disallowed_methods)]
+	fn get_price_base_asset(
+		(base_asset_reserves, quote_asset_reserves, peg_multiplier) in min_max_reserve(),
+	) {
+		ExtBuilder {
+			vamm_count: 1,
+			vamms: vec![
+				(0,
+				 VammState{
+					 base_asset_reserves,
+					 quote_asset_reserves,
+					 peg_multiplier,
+					 invariant: Default::default(),
+					 closed: None})]
+		}.build().execute_with(|| {
+			let quote_peg = quote_asset_reserves.checked_mul(peg_multiplier);
+
+			if quote_peg.is_none() {
+				assert_noop!(
+					TestPallet::get_price(0, AssetType::Base),
+					ArithmeticError::Overflow);
+			} else if quote_peg.unwrap().checked_div(base_asset_reserves).is_none() {
+				assert_noop!(
+					TestPallet::get_price(0, AssetType::Base),
+					ArithmeticError::DivisionByZero);
+			} else {
+				assert_ok!(
+					TestPallet::get_price(0, AssetType::Base),
+					Decimal::from_inner(quote_peg.unwrap().checked_div(base_asset_reserves).unwrap()))
```

Comment:
I think you should convert `quote_peg` and `base_asset_reserves` to `Decimal` **before** the division, no?
Can you check that doing the above vs. what you're doing now yields the same results? Try with `quote_peg` being half of `base_asset_reserves` and check if the price is `0.5`

Reply:
Interesting enough both this version and the one converting to `Decimal` before handle the same results (all tests passes for both of them...)

I'll need to investigate if further next week than. 

Thank you for the comment on this!
I'll create a new task for it on clickup.

Reply:
I really want to thank you for raising attention to this section. 

Yeah, the values should be converted in `Decimal` before the computation, but the problem of both `Decimal` and `u128` passing all tests was due to the fact that the range of valid input values for the variables was too permissive, so the fuzz tests didn't really catch the issue with the `Decimal`s. 

I fixed this error and also create individual tests for each specific case. (something in the lines of [this comment here](https://github.com/ComposableFi/composable/pull/1013#discussion_r874066000))

Thanks again! :)